src/cli/rustup_mode.rs

```diff
@@ -1052,11 +1084,9 @@ fn which(cfg: &Cfg, m: &ArgMatches) -> Result<utils::ExitCode> {
 }
 
 #[cfg_attr(feature = "otel", tracing::instrument(skip_all))]
-fn show(cfg: &Cfg, m: &ArgMatches) -> Result<utils::ExitCode> {
+fn show(cfg: &Cfg, verbose: bool) -> Result<utils::ExitCode> {
```

Comment:
> IMO it would be nice to take care of these changes in a separate commit, since it's pretty much independent of the change to derive-based argument parsing.

I'm not quite sure. Changes like this are a direct consequence of no longer having access to `&ArgMatches` in the `clap_derive` mode. Also, these changes are quite mechanical, in that these parameters were actually there in the original code, it's just that they were usually located at the first lines of the function body instead of the signature line. For example, in this particular function, the change to use `verbose: bool` corresponds to:

https://github.com/rust-lang/rustup/blob/66b7a442c074c5cd597ea1436e1bafb0807b46e0/src/cli/rustup_mode.rs#L1064-L1067

Thus, I don't see too much value separating it into a new commit.

Reply:
IMO the reason this makes sense is that the `ArgMatches` approach creates an unnecessary coupling between the CLI frontend and the core functionality of the tools. One reason clap's derive macros are good is that they decrease the amount of coupling, but IMO it's still independently a good idea to decrease coupling in order to make the codebase more modular.

For example, if we ever wanted to offer a library interface we wouldn't want to have these `ArgMatches` arguments in there. As such, IMO it's really bad design to have these functions take `ArgMatches` directly, and correcting that makes sense even without the migration to derive macros.

Reply:
@djc Thanks a lot for your clarification, I can see what you mean now. I'm really sorry but I don't have enough bandwidth recently to spend on this PR (like I did back in December '23), so I guess I'll probably leave it as it is.